utils/pindialer/dialer.go

```diff
@@ -21,11 +21,13 @@ func MakeDialer(fingerprint string) Dialer {
 		}
 		connstate := c.ConnectionState()
 		for _, chain := range connstate.VerifiedChains {
-			leafCert := chain[0]
-			hash := sha256.Sum256(leafCert.RawSubjectPublicKeyInfo)
-			digest := base64.StdEncoding.EncodeToString(hash[:])
-			if digest == fingerprint {
-				return c, nil
+			// allow for intermediate certificate pinning, or leaf certificate pinning
+			for _, node := range chain {
```

Comment:
Since we're checking the chain, is there something we need to do to ensure we don't validate expired certs?

Reply:
The tls.Dial takes care of checking for expired certificates, and other validity checks.  We only care that the particular certificate that we know is good is in the verified chain.